lib/Pipeline/ExecutionContext.cpp

```diff
@@ -0,0 +1,56 @@
+/// \file ExecutionContext.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/ExecutionContext.h"
+#include "revng/Pipeline/Step.h"
+#include "revng/Pipeline/Target.h"
+
+using namespace pipeline;
+
+ExecutionContext::ExecutionContext(Context &Ctx,
+                                   Step &Step,
+                                   PipeWrapper *Pipe) :
+  TheContext(&Ctx),
+  CurrentStep(&Step),
+  Pipe(Pipe),
+  RunningOnPipe(Pipe != nullptr) {
+  // pipe is null when execution a analysis.
+  if (RunningOnPipe)
```

Comment:
i went for this solution because otherwise it would overcomplicate the invokable.h logic. It can be done, but there is no much point to do so, and surely at some point one will wish to get some run time dependent information from the execution context, such as which step is the current one.

Reply:
Add a comment about the fact that, as of now, we could just use `Context`.